### PR TITLE
Make lfs bootable

### PR DESCRIPTION
**Pull request by travis1829**
Booting with lfs now works.
* Most basic user programs (ls, cat, grep, wc, etc.) work fine.
* Reading anything (inode, data) seems to work fine, but trying to write something seems to sometimes causes a problem.
-----
* kernel booting
  * kernel main: O
  * superblock load: O
  * checkpoint load: O
  * imap load: O
  * inode, data block load: O
* user program init
  * exec: O
  * run sh: O
* user programs
  * cat O
  * echo O
  * forktest O
  * grep O
  * init O
  * ln ~~X~~ -> O
  * ls O
  * mkdir ~~X~~ -> O
  * rm ~~X~~ -> O
  * sh O
  * stressfs X
  * usertests X
    * most tests not related to fs seem to work
  * grind X
  * wc O
  * zombie O
  * hello O

To run, do
```
make clean
FS=lfs CPUS=1 make qemu
```

### CODE REVIEW

File: kernel-rs/src/fs/lfs/inode.rs [hunk 1]
```diff
@@ -256,40 +256,87 @@ impl InodeGuard<'_, Lfs> {
...
+                    // Copy from old block to new block.
+                    let (mut buf, new_addr) = tx
+                        .fs
+                        .segment()
+                        .get_or_add_data_block(inum, bn as u32, ctx)
+                        .unwrap();
+                    let old_buf = hal().disk().read(dev, addr, ctx);
+                    // SAFETY: The old data block's content will not be used from now on.
+                    unsafe {
+                        core::ptr::copy(
```
Comment (yj-han) [inline kernel-rs/src/fs/lfs/inode.rs]: Does this code mean that data inside `old_buf` is written on `buf`? Is it equivalent to write something on hal().disk()?

**Comment by yj-han** [comment]
Will this code pass if we add tests with `features=lfs` to GitHub CI?

File: kernel-rs/src/fs/lfs/inode.rs [hunk 2]
```diff
@@ -200,18 +200,15 @@ impl InodeGuard<'_, Lfs> {
             .get_or_add_updated_inode_block(self.inum, ctx)
             .unwrap();
 
-        const_assert!(IPB <= mem::size_of::<BufData>() / mem::size_of::<Dinode>());
+        const_assert!(mem::size_of::<BufData>() >= mem::size_of::<Dinode>());
```
[the remark this answers is not in the document]
Comment (travis1829) [reply kernel-rs/src/fs/lfs/inode.rs]: Fixed.

File: kernel-rs/src/fs/lfs/inode.rs (hunk 1 above)
Comment (travis1829) [reply kernel-rs/src/fs/lfs/inode.rs]: Yes, the data on `old_buf` is just being copied to the `buf` here.
You can think of `ptr::copy` something similar to `memcpy` in C.

**Comment by travis1829** [comment]
> Will this code pass if we add tests with `features=lfs` to GitHub CI?

Currently, the rv6 with lfs will pass only a few of usertests related to file system, so I think we should add CI checks later.

**Pull request merged**